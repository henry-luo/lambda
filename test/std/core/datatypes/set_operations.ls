// Test: Set Operations
// Category: core/datatypes
// Type: positive
// Expected: 1

// Set literals (if supported, otherwise create from arrays)
let set1 = new Set([1, 2, 3, 4, 5])
let set2 = new Set([4, 5, 6, 7, 8])

// Basic operations
set1.add(6)         // Add element
let has = set1.has(3)  // true
let size = set1.size   // 6
set1.delete(1)       // Remove element

// Set operations
let union = new Set([...set1, ...set2])  // Union
let intersection = new Set([...set1].filter(x => set2.has(x)))  // Intersection
let difference = new Set([...set1].filter(x => !set2.has(x)))   // Difference

// Convert to array
let array_from_set = [...set1]  // [2, 3, 4, 5, 6]

// Final check
1
