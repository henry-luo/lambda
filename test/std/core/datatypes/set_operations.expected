09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/set_operations.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 7, Col 12 - 7, Col 15: ERROR
09:15:42 [ERROR] ERROR node at Ln 7, Col 12 - 7, Col 15
09:15:42 [ERROR] Syntax error at Ln 8, Col 12 - 8, Col 15: ERROR
09:15:42 [ERROR] ERROR node at Ln 8, Col 12 - 8, Col 15
09:15:42 [ERROR] Syntax error at Ln 17, Col 13 - 17, Col 16: ERROR
09:15:42 [ERROR] ERROR node at Ln 17, Col 13 - 17, Col 16
09:15:42 [ERROR] Syntax error at Ln 17, Col 21 - 17, Col 22: ERROR
09:15:42 [ERROR] ERROR node at Ln 17, Col 21 - 17, Col 22
09:15:42 [ERROR] Syntax error at Ln 17, Col 38 - 17, Col 39: ERROR
09:15:42 [ERROR] ERROR node at Ln 17, Col 38 - 17, Col 39
09:15:42 [ERROR] Syntax error at Ln 18, Col 20 - 18, Col 23: ERROR
09:15:42 [ERROR] ERROR node at Ln 18, Col 20 - 18, Col 23
09:15:42 [ERROR] Syntax error at Ln 18, Col 29 - 18, Col 32: ERROR
09:15:42 [ERROR] ERROR node at Ln 18, Col 29 - 18, Col 32
09:15:42 [ERROR] Syntax error at Ln 18, Col 45 - 18, Col 49: ERROR
09:15:42 [ERROR] ERROR node at Ln 18, Col 45 - 18, Col 49
09:15:42 [ERROR] Syntax error at Ln 19, Col 18 - 19, Col 21: ERROR
09:15:42 [ERROR] ERROR node at Ln 19, Col 18 - 19, Col 21
09:15:42 [ERROR] Syntax error at Ln 19, Col 27 - 19, Col 30: ERROR
09:15:42 [ERROR] ERROR node at Ln 19, Col 27 - 19, Col 30
09:15:42 [ERROR] Syntax error at Ln 19, Col 45 - 19, Col 47: ERROR
09:15:42 [ERROR] ERROR node at Ln 19, Col 45 - 19, Col 47
09:15:42 [ERROR] Syntax error at Ln 22, Col 23 - 22, Col 26: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 23 - 22, Col 26
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/set_operations.ls' executed: #####
null
